crates/biome_html_parser/src/syntax/svelte.rs

```diff
@@ -1,20 +1,98 @@
 use crate::parser::HtmlParser;
-use crate::syntax::parse_error::expected_svelte_closing_block;
+use crate::syntax::parse_error::{expected_child, expected_svelte_closing_block};
+use crate::syntax::{TextExpression, parse_html_element};
 use crate::token_source::HtmlLexContext;
 use biome_html_syntax::HtmlSyntaxKind::{
-    EOF, SVELTE_BINDING_LIST, SVELTE_BOGUS_BLOCK, SVELTE_DEBUG_BLOCK, SVELTE_IDENT, SVELTE_NAME,
+    EOF, HTML_BOGUS_ELEMENT, HTML_ELEMENT_LIST, SVELTE_BINDING_LIST, SVELTE_BOGUS_BLOCK,
+    SVELTE_DEBUG_BLOCK, SVELTE_IDENT, SVELTE_KEY_BLOCK, SVELTE_KEY_CLOSING_BLOCK,
+    SVELTE_KEY_OPENING_BLOCK, SVELTE_NAME,
 };
 use biome_html_syntax::{HtmlSyntaxKind, T};
-use biome_parser::parse_lists::ParseSeparatedList;
+use biome_parser::parse_lists::{ParseNodeList, ParseSeparatedList};
 use biome_parser::parse_recovery::{ParseRecoveryTokenSet, RecoveryResult};
 use biome_parser::prelude::ParsedSyntax;
 use biome_parser::prelude::ParsedSyntax::{Absent, Present};
 use biome_parser::{Marker, Parser, TokenSet, token_set};
 
+pub(crate) fn parse_svelte_hash_block(p: &mut HtmlParser) -> ParsedSyntax {
+    if !p.at(T!["{#"]) {
+        return Absent;
+    }
+    // NOTE: use or_else chain here to parse
+    // other possible hash blocks
+    parse_key_block(p)
```

Comment:
The reason why we can't use `match p.cur()` is because svelte keywords are lexed only under certain context, so we can't use `p.nth()` with a different context.

So in order to parse other blocks, we use `or_else`. Let me know if there's a better way